src/AsyncEventSource.cpp

```diff
@@ -276,56 +294,70 @@ void AsyncEventSource::_addClient(AsyncEventSourceClient * client){
     client->write((const char *)temp, 2053);
     free(temp);
   }*/
-  
+  _client_queue_lock.lock();
```

Comment:
By calling the connect callback with the client queue lock held, this code introduces a self-deadlock if the connect callback, in turn, tries to broadcast a message to the event source using AsyncEventSource::send (or tries to use any other public API that tries to take that very same lock).

I had this exact scenario in my project, and I had to fix it as shown in commit https://github.com/yubox-node-org/ESPAsyncWebServer/commit/646eab7bd8761a7ff89c627e37c0f27c49cef32f of yuboxfixes branch of the ESPAsyncWebServer fork at yubox-node-org.

Reply:
I just now realized that the extern void *pxCurrentTCB member of class AsyncWebLock does /not/ point to the TCP TCB struct (TCB as in "transfer control block") but to the FreeRTOS stack pointer, and that this is how the AsyncWebLockGuard works..
I only did minimal testing, but I assume this is safe to pull into this PR for completeness.

I hope that the original author does find some time in the near future to go through all these forks and PRs...
Thank you very much for the notice!